### PR TITLE
Perl tidy

### PR DESCRIPTION
**Pull request by wingolab**
**Major changes** were to organize the perl code, tidy perl code, add tooling for easy tidying, add docker recipe, and add github workflows for perl code tidying and packaging.

**Minor changes** include updates to tests to eliminate minor errors and specifying DBD::mysql version.

Suggestions for the future include
1. update documentation for installation in a separate pull request
2. eliminating node, pm2, and awscli from the docker recipe that focuses on the Perl package and move those to a separate docker recipe for the server
3. consider consolidating installation scripts or transition to using docker for ease of maintenance 
4. transition system calls in the Perl tests to use Perl-based tooling

### CODE REVIEW

File: perl/README.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+Assuming that you've cloned the repository and are working on it locally, then the dependencies can mostly be installed with cpanm.
+But there are a few one-off dependencies that require a slightly modified approach.
+
+One-off dependencies can be installed as follows:
+
+```bash
+cpanm --quiet https://github.com/bystrogenomics/msgpack-perl.git
+cpanm --quiet --notest MouseX::Getopt
+git clone --depth 1 -branch v0.12 --recurse-submodules https://github.com/salortiz/LMDB_File.git \
```
Comment (akotlar) [inline perl/README.md:21]: -branch v0.12 should be `--branch v0.12`

File: perl/README.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  && cd LMDB_File \
+  && cpanm --quiet .
```
Comment (akotlar) [inline perl/README.md:23]: needs` \` at end of line

File: perl/README.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  && cd .. \
+  && rm -rf LMDB_File
+cpanm --quiet DBD::mysql@4.051
+```
+
+The remaining dependencies are installed like this:
+
+```bash
+cpanm --installdeps .
```
Comment (akotlar) [inline perl/README.md:32]: This doesn't work:

```
You have /usr/bin/unzip
--> Working on .
Entering /mnt/ssd1/bystro/perl
Configuring /mnt/ssd1/bystro/perl
-> N/A
! Configuring . failed. See /home/alexkotlar/.cpanm/work/1697325102.82941/build.log for details.
```

I think maybe lack of Makefile?

**Review by akotlar — CHANGES_REQUESTED** [review]
A few changes needed in README.md

File: perl/README.md (hunk 3 above)
Comment (wingolab) [reply perl/README.md:35]: Lots of ways to achieve this, but adding the cpanfile makes it very clear the intention, I think.

File: perl/README.md (hunk 1 above)
Comment (wingolab) [reply perl/README.md]: Let's specify versions in the perl code itself at which point this could be updated. Right now, I dropped the branch switch.

**Comment by akotlar** [comment]
cpanm install now works great; dzil install fails because it doesn't like an existing cpanfile. Is it possible to skip cpanfile creation, or remove the cpanfile before dzil install (it seems dzil install adds it back as a matter of course)

**Review by akotlar — CHANGES_REQUESTED** [review]
Needs fixed dzill install (cpanfile conflict) and then good to go

**Comment by akotlar** [comment]
Thanks for addressing the comments!

**Pull request merged**